ext/gmp/gmp.c

```diff
@@ -1373,13 +1385,14 @@ ZEND_FUNCTION(gmp_root)
 	}
 
 	if (nth <= 0) {
-		php_error_docref(NULL, E_WARNING, "The root must be positive");
-		RETURN_FALSE;
+		zend_argument_value_error(2, "must be greater than or equal to 1");
```

Comment:
it's a bit simpler, so probably worth considering:
```suggestion
		zend_argument_value_error(2, "must be greater than 0");
```